helper/ResponseVariableFormatter.php

```diff
@@ -0,0 +1,119 @@
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2017 (original work) Open Assessment Technologies SA (under the project TAO-PRODUCT);
+ *
+ */
+
+namespace oat\taoOutcomeUi\helper;
+
+use \taoResultServer_models_classes_ResponseVariable as ResponseVariable;
+
+/**
+ * The ResponseVariableFormatter enables you to format the output of the ResultsService into an associative compatible with the client code
+ * Class ResponseVariableFormatter
+ * @package oat\taoOutcomeUi\helper
+ */
+class ResponseVariableFormatter {
+
+    /**
+     * Special trim for identifier in response variables
+     * @param $identifierString
+     * @return string
+     */
+    static private function trimIdentifier($identifierString){
+        return trim($identifierString, " \t\n\r\0\x0B'\"");
+    }
+
+    /**
+     * Format a string into the appropriate format according to the base type
+     * @param $baseType
+     * @param $stringValue
+     * @return array|bool|float|int|string
+     * @throws \common_Exception
+     */
+    static private function formatStringValue($baseType, $stringValue){
+
+        switch(strtolower($baseType)){
+            case 'string':
+            case 'duration':
+                return $stringValue;
+            case 'identifier':
+                return self::trimIdentifier($stringValue);
+            case 'integer':
+                return intval($stringValue);
+            case 'float':
+                return floatval($stringValue);
+            case 'boolean':
+                return (trim($stringValue) === 'true' || $stringValue === true) ? true : false;
+            case 'pair':
+            case 'directedpair':
+                $pair = explode(' ', trim($stringValue));
+                if(count($pair) != 2){
+                    throw new \common_Exception('invalid pair string');
+                }
+                return [self::trimIdentifier($pair[0]), self::trimIdentifier($pair[1])];
+            case 'point':
+                $pair = explode(' ', trim($stringValue));
+                if(count($pair) != 2){
+                    throw new \common_Exception('invalid point string');
+                }
+                return [intval($pair[0]), intval($pair[1])];
+            default:
+                throw new \common_Exception('unknown basetype');
+        }
+    }
+
+    /**
+     * Format a standard ResponseVariable into a associative array, directly usable on the client side.
+     * The common usage is to use this to format the output of oat\taoOutcomeUi\model\ResultsService::getStructuredVariables();
+     *
+     * @param ResponseVariable $var
+     * @return array
+     * @throws \common_Exception
+     */
+    static public function formatVariableToPci(ResponseVariable $var) {
+        $value = base64_decode($var->getValue());
```

Comment:
Even with the Base64 check I still have corrupted data :(
Is actually the `base64_decode()` mandatory? Could we apply it only on particular variable type?